server/util/prefix/prefix.go

```diff
@@ -20,9 +21,14 @@ func addPrefix(prefix, key string) string {
 
 func userPrefixCacheKey(ctx context.Context, env environment.Env, key string) (string, error) {
 	if auth := env.GetAuthenticator(); auth != nil {
-		if u, err := auth.AuthenticatedUser(ctx); err == nil {
-			if u.GetGroupID() == "" {
-				return "", status.PermissionDeniedErrorf("Attempting to write to cache as a user with no group.")
+		u, err := auth.AuthenticatedUser(ctx)
+		if err != nil {
+			if authutil.IsInvalidCredentialsError(err) {
+				return "", err
+			}
+		} else {
+			if u.GetGroupID() == "" && !env.GetAuthenticator().AnonymousUsageEnabled() {
```

Comment:
If we get here, doesn't it mean the user is not anonymous? In which case it makes sense to keep the old check, otherwise we will could call addPrefix with an empty group ID.

Reply:
Reverted back to the old logic in this case.